spec/controllers/ems_common_controller_spec.rb

```diff
@@ -83,15 +83,20 @@
     context "#update_button_validate" do
```

Comment:
It would nice to do whatever we need to be able to test that newly extracted method, outside of the controller spec... Maybe the query method should live in a helper or something but it should be testable to try to catch any bugs when we need to modify those many conditionals.